backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,502 @@
+	// TODO: Add resilience to resource version changing
+	err = k.client.Status().Update(context.TODO(), k8sPipelineVersion)
+	if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline version status")
+	}
+
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineVersion(pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	pipeline, err := k.GetPipeline(pipelineVersion.PipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.createPipelineVersionWithPipeline(pipeline, pipelineVersion)
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
+		"pipeline default version is unsupported when storing in Kubernetes")
+}
+
+func (k *PipelineStoreKubernetes) GetLatestPipelineVersion(pipelineId string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if !k8sPipelineVersion.IsOwnedByPipeline(pipelineId) {
+			continue
+		}
+
+		if latestK8sPipelineVersion == nil || k8sPipelineVersion.CreationTimestamp.Time.After(latestK8sPipelineVersion.CreationTimestamp.Time) {
+			latestK8sPipelineVersion = &k8sPipelineVersion
+
+			continue
+		}
+	}
+
+	if latestK8sPipelineVersion == nil {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", "Latest")
+	}
+
+	return latestK8sPipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersion(pipelineVersionId string) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.getK8sPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionByName(name string) (*model.PipelineVersion, error) {
+	// TODO: Do we need this?
+	return nil, nil
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionWithStatus(pipelineVersionId string, status model.PipelineVersionStatus) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.GetPipelineVersion(pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	if pipelineVersion.Status != status {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", pipelineVersionId)
+	}
+
+	return pipelineVersion, nil
+}
+
+func (k *PipelineStoreKubernetes) ListPipelineVersions(pipelineId string, opts *list.Options) (versions []*model.PipelineVersion, totalSize int, nextPageToken string, err error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions()
+	if err != nil {
+		return nil, 0, "", err
+	}
+
+	pipelineVersions := make([]*model.PipelineVersion, 0, len(k8sPipelineVersions.Items))
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if k8sPipelineVersion.IsOwnedByPipeline(pipelineId) {
```

Comment:
Let's use a label selector to filter the pipeline versions to just those by this pipeline ID. The mutating webhook will set a label for this by default, but you can have the REST API set it when uploading a pipeline version for now.

Reply:
@rimolive could you look at this comment again please?